src/rpc/blockchain.cpp

```diff
@@ -1526,7 +1535,14 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
     UniValue ret(UniValue::VOBJ);
     ret.push_back(Pair("time", (int64_t)pindex->nTime));
     ret.push_back(Pair("txcount", (int64_t)pindex->nChainTx));
-    ret.push_back(Pair("txrate", ((double)nTxDiff) / nTimeDiff));
+    if (blockcount > 0) {
+        ret.push_back(Pair("block_count", blockcount));
+        ret.push_back(Pair("txratecount", nTxDiff));
+        ret.push_back(Pair("txrateinterval", nTimeDiff));
+    }
+    if (nTimeDiff > 0) {
```

Comment:
Perhaps move this up into the `if (blockcount > 0)` code block (since `nTimeDiff` can only be non-zero when `blockcount` > 0)? Up to you.